src/PricingModules/UniswapV3/UniswapV3WithFeesPricingModule.sol

```diff
@@ -0,0 +1,442 @@
+    using FixedPointMathLib for uint256;
+    using FullMath for uint256;
+
+    /* //////////////////////////////////////////////////////////////
+                                STORAGE
+    ////////////////////////////////////////////////////////////// */
+
+    // Map asset => uniswapV3Factory.
+    mapping(address => address) public assetToV3Factory;
+    // Map asset => assetUnit.
+    mapping(address => uint256) public unit;
+
+    // The Arcadia Pricing Module for standard ERC20 tokens (the underlying assets).
+    PricingModule immutable erc20PricingModule;
+
+    /* //////////////////////////////////////////////////////////////
+                                CONSTRUCTOR
+    ////////////////////////////////////////////////////////////// */
+
+    /**
+     * @param mainRegistry_ The contract address of the MainRegistry.
+     * @param oracleHub_ The contract address of the OracleHub.
+     * @param riskManager_ The address of the Risk Manager.
+     * @dev AssetType for Uniswap V3 Liquidity Positions (ERC721) is 1.
+     */
+    constructor(address mainRegistry_, address oracleHub_, address riskManager_, address erc20PricingModule_)
+        PricingModule(mainRegistry_, oracleHub_, 1, riskManager_)
+    {
+        erc20PricingModule = PricingModule(erc20PricingModule_);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        ASSET MANAGEMENT
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Adds a new asset to the UniswapV3PricingModule.
+     * @param asset The contract address of the asset (also known as the NonfungiblePositionManager).
+     * @dev Per protocol (eg. Uniswap V3 and its forks) there is a single asset,
+     * and each liquidity position will have a different id.
+     */
+    function addAsset(address asset) external onlyOwner {
+        require(!inPricingModule[asset], "PMUV3_AA: already added");
+
+        inPricingModule[asset] = true;
+        assetsInPricingModule.push(asset);
+
+        assetToV3Factory[asset] = INonfungiblePositionManager(asset).factory();
+
+        // Will revert in MainRegistry if asset can't be added.
+        IMainRegistry(mainRegistry).addAsset(asset, assetType);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        ALLOW LIST MANAGEMENT
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Checks for a token address and the corresponding Id if it is allow-listed.
+     * @param asset The contract address of the asset.
+     * @param assetId The Id of the asset.
+     * @return A boolean, indicating if the asset is whitelisted.
+     */
+    function isAllowListed(address asset, uint256 assetId) public view override returns (bool) {
+        if (!inPricingModule[asset]) return false;
+
+        try INonfungiblePositionManager(asset).positions(assetId) returns (
+            uint96,
```

Comment:
just use comma's instead of the asset type

Reply:
Not possible in a try, catch